src/utils/data.js

```diff
@@ -55,8 +56,7 @@ export async function fetchData(path, params = {}) {
 }
 
 function createDataPath(path) {
-  const dataPath = path.replace(/\/$/, '');
-  return `${pathJoin([baseUrl, 'data', dataPath])}.json`;
+  return `${pathJoin(['/data', path.replace(/\/$/, '')])}.json`;
```

Comment:
by removing the `baseUrl` here, wont this break deployments to none root paths? Like https://ethan-kusters.github.io/swift-argument-parser/documentation/argumentparser/

Reply:
Not really because I'm using `getAbsoluteUrl` in https://github.com/apple/swift-docc-render/pull/357/files#diff-3cf45312eb60039df9dfa67d90bc0936e830f0d3424cf0b09943b1e3b09d700fR34 which contains the baseUrl on it. Before we weren't adding the baseUrl inside the `fetchData` function but now we do.

Reply:
mmmm no, thats not correct.

https://github.com/apple/swift-docc-render/pull/357/files#diff-3cf45312eb60039df9dfa67d90bc0936e830f0d3424cf0b09943b1e3b09d700fR34 - this creates a fully qualified URL from a path, but for the most part, that just sticks the path to whatever the current `window.location.href` is. Since our paths start with `/` it will remove the entire `pathname` from the current `location.href`.

The path we pass is transformed by `createDataPath` which was adding the baseUrl prefix, which you removed. 

Here you can see what I mean, you are essentially introducing a bug with deployments to subpaths.

![image](https://user-images.githubusercontent.com/9863944/174728374-d98c1919-f436-4f8e-b7d1-a88392e31e3b.png)

Reply:
I believe there is a misunderstanding on the `url-helper.js` file since the param for the domain path was also called `baseURL` but it's just the `domainPath`. I changed it here: https://github.com/apple/swift-docc-render/pull/357/commits/52827fbd6f966bee5a16e4ae801b31a9ccfedea6

I didn't remove the baseURL prefix, it happens in the `normalizeUrl` function inside `getAbsoluteUrl`.

The baseURL prefix happens inside the `path` const and in the case of `theme-settings.json` it's a file that is hosted in the source root so it makes sense to delete the rest of the pathname in the window.location.href

<img width="566" alt="Screenshot 2022-06-21 at 15 38 08" src="https://user-images.githubusercontent.com/8567677/174813660-c4d99cba-4d09-46a8-bee9-609e179182c1.png">